### DISCUSSION TITLE
Autokeras + TF + TF datasets on GPU

### DISCUSSION THREAD

**Discussion by JuliaWasala** [keras-team/autokeras#1669] [Q&A]
Hi there,

I've been having issues with finding exact versions of Autokeras, TF + TF datasets that can work in a conda env to run code on the GPU.

The installation guide on the website did not work for me, the GPUs would not get properly recognized.
At first, I was able to run example code from the website for using AutoModel on a GPU I have access to by creating a conda env, installing: `conda install -c nvidia cudnn=8.2.1` and then installing autokeras directly.
However, when I want to use another dataset from TFDF (e.g. DIV2K) I noticed that after installation of tensorflow-datasets ,`pip install tensorflow-datasets`,  I can't run the example code at all anymore. The error I get the same error as in this thread (https://github.com/keras-team/autokeras/issues/1326): The dataset should at least contain 2 batches to be split.

I was using autokeras 1.0.16.post1
tensorflow 2.5.2 
keras-tuner 1.0.4
tensorflow-datasets 4.4.0
python 3.9.7

This puzzled me, because in the autokeras 1.0.16.post1 release it is said that the TF version is pinned to 2.5.0, so why is 2.5.2 installing ?

Then, I decided to try something else. I remove my env, make a new one, install cudnn. Then I install tensorflow 2.5.0, autokeras 1.0.16. and tensorflow-datasets with no version specified because I can not find anything about version compatibilities.

Now I get the following error when running example: 
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_393/867679257.py in <module>
      7 output_node = ak.ClassificationHead()(output_node)
      8 
----> 9 auto_model = ak.AutoModel(
     10     inputs=input_node, outputs=output_node, overwrite=True, max_trials=1
     11 )

/local/s1620444/automl-sr-rs-env/lib/python3.9/site-packages/autokeras/auto_model.py in __init__(self, inputs, outputs, project_name, max_trials, directory, objective, tuner, overwrite, seed, max_model_size, **kwargs)
    140         if isinstance(tuner, str):
    141             tuner = get_tuner_class(tuner)
--> 142         self.tuner = tuner(
    143             hypermodel=graph,
    144             overwrite=overwrite,

/local/s1620444/automl-sr-rs-env/lib/python3.9/site-packages/autokeras/tuners/greedy.py in __init__(self, hypermodel, objective, max_trials, initial_hps, seed, hyperparameters, tune_new_entries, allow_new_entries, **kwargs)
    228             allow_new_entries=allow_new_entries,
    229         )
--> 230         super().__init__(oracle=oracle, hypermodel=hypermodel, **kwargs)

/local/s1620444/automl-sr-rs-env/lib/python3.9/site-packages/autokeras/engine/tuner.py in __init__(self, oracle, hypermodel, **kwargs)
     52         super().__init__(oracle, hypermodel, **kwargs)
     53         # Save or load the HyperModel.
---> 54         self.hypermodel.hypermodel.save(os.path.join(self.project_dir, "graph"))
     55         self.hyper_pipeline = None
     56 

AttributeError: 'Graph' object has no attribute 'hypermodel'
```

I find another github page about this, where they recommend to downgrade keras-tuner to 1.0.4 (because this time, keras-tuner 1.1.0 was installed), but that doesnt change anything.

TL;DR
I was using these package versions to run autokeras on GPU: I was using autokeras 1.0.16.post1
tensorflow 2.5.2 
keras-tuner 1.0.4
python 3.9.7

But tensorflow-datasets 4.4.0 broke it, making it impossible to use automodel at all. Which versions should I use to use these packages together on GPU?
